bindings/python/tests/compiler_core_test.py

```diff
@@ -0,0 +1,136 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import os
+import io
+import tempfile
+import unittest
+
+from pyiree import compiler2 as compiler
+
+SIMPLE_MUL_ASM = """
+func @simple_mul(%arg0: tensor<4xf32>, %arg1: tensor<4xf32>) -> tensor<4xf32>
+      attributes { iree.module.export } {
+    %0 = "mhlo.multiply"(%arg0, %arg1) {name = "mul.1"} : (tensor<4xf32>, tensor<4xf32>) -> tensor<4xf32>
+    return %0 : tensor<4xf32>
+}
+"""
+
+
+class CompilerTest(unittest.TestCase):
+
+  def testNoTargetBackends(self):
+    with self.assertRaisesRegex(
+        ValueError, "Expected a non-empty list for 'target_backends'"):
+      binary = compiler.compile_str(SIMPLE_MUL_ASM)
+
+  def testCompileStr(self):
+    binary = compiler.compile_str(
+        SIMPLE_MUL_ASM, target_backends=compiler.DEFAULT_TESTING_BACKENDS)
+    print("Flatbuffer size =", len(binary))
```

Comment:
It would be nice to find/replace `print`--> `logging.info` here, as IME `print` statements can appear wildly out of order in test cases for some reason (e.g. all print statements pooled at the bottom in `bindings/python/pyiree/rt/vm_test.py` when I executed it (with `bazel`)).